src/Microsoft.Diagnostics.Monitoring/DiagnosticsEventPipeProcessor.cs

```diff
@@ -455,6 +468,40 @@ private async Task HandleGCEvents(EventPipeEventSource source, int pid, Func<Tas
             _gcGraph.AllowReading();
         }
 
+        private async Task HandleProcessInfo(EventPipeEventSource source, Func<Task> stopFunc, CancellationToken token)
+        {
+            string commandLine = null;
+
+            Action<TraceEvent, Action> processInfoHandler = (TraceEvent traceEvent, Action taskComplete) =>
+            {
+                commandLine = (string)traceEvent.PayloadByName("CommandLine");
+                taskComplete();
+            };
+
+            using var processInfoTaskSource = new EventTaskSource<Action<TraceEvent>>(
```

Comment:
For cases where we just need this info:
- What's the general perf?
- Can we omit rundown/use a very small buffer?
- Do you get the data even in forward connections, or do they have to be reversed?

Reply:
If I recall correctly, we should be able to omit rundown. I'll try a smaller buffer. This data can be gotten for both connections (basically, the entire diagnostics protocol and behavior (except for the socket/pipe itself) is agnostic of forward vs reversed.

Reply:
> I've been looking at adding a command to the diagnostics IPC that would allow you to query this data, e.g., runtime cookie, commandline, etc., without setting up an EventPipe session. I'm assuming that would be preferable to this scheme?

::I'm still not fully familiar how the GitHub commenting works, so I'm copying my command below back to this thread::

@josalem, absolutely! Something more lightweight than using an event pipe session would be much more preferable.

Reply:
I've replaced this code with querying IIpcEndpoint.CheckTransport since that now verifies that the existing stream is still connected with the runtime instance. However, I'd still like the command to get the process information separately from using an event pipe session since we'll be using that to populate more process information in the /processes HTTP endpoint in dotnet-monitor.